rsocket-ipc-graphql/src/main/java/io/rsocket/graphql/GraphQLServerRequestResponse.java

```diff
@@ -0,0 +1,91 @@
+package io.rsocket.graphql;
+
+import graphql.ExecutionInput;
+import graphql.ExecutionResult;
+import graphql.GraphQL;
+import graphql.GraphQLError;
+import graphql.execution.ExecutionId;
+import graphql.execution.instrumentation.Instrumentation;
+import graphql.schema.GraphQLSchema;
+import io.netty.buffer.ByteBuf;
+import io.rsocket.ipc.Functions;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import org.dataloader.DataLoaderRegistry;
+import org.reactivestreams.Publisher;
+import reactor.core.publisher.Mono;
+
+@SuppressWarnings("unchecked")
+class GraphQLServerRequestResponse implements Functions.RequestResponse<GraphQLRequest, Object> {
+  private final DataLoaderRegistry registry;
+  private final Instrumentation instrumentation;
+  private final GraphQLSchema graphQLSchema;
+
+  GraphQLServerRequestResponse(
+      DataLoaderRegistry registry, Instrumentation instrumentation, GraphQLSchema graphQLSchema) {
+    this.registry = registry;
+    this.instrumentation = instrumentation;
+    this.graphQLSchema = graphQLSchema;
+  }
+
+  @Override
+  public Mono<Object> apply(GraphQLRequest request, ByteBuf byteBuf) {
+    try {
+      ExecutionInput.Builder builder =
+          ExecutionInput.newExecutionInput()
+              .query(request.getQuery())
+              .operationName(request.getOperationName())
+              .variables(request.getVariables())
+              .context(byteBuf)
+              .executionId(ExecutionId.generate());
+
+      if (registry != null) {
+        builder.dataLoaderRegistry(registry);
+      }
+
+      ExecutionInput executionInput = builder.build();
+
+      CompletableFuture<ExecutionResult> result =
+          GraphQL.newGraphQL(graphQLSchema)
+              .instrumentation(instrumentation)
+              .build()
+              .executeAsync(executionInput);
+
+      return Mono.fromFuture(result)
+          .flatMap(
+              executionResult -> {
+                List<GraphQLError> errors = executionResult.getErrors();
+
+                if (!errors.isEmpty()) {
+                  return Mono.error(GraphQLErrorException.from(errors));
+                }
+
+                if (!executionResult.isDataPresent()) {
```

Comment:
Not sure if you can extract a reuse this functionality between the streams (subscription) and request response operations (query, mutate)?

But it feels a bit caught in the middle.  Either

1) Make this a utility so when you read this you can skip over this complexity of the GraphQL-Java quirks
or
2) Be more definitive and fail except when you get the publisher or mono you expect
or
3) Explain why and how each branch is needed, bad API?

Reply:
Is there some external choice that affects whether you get Publisher (generic) or Mono (reactor?)

Reply:
The API is kind of clunky, but my understanding is that you can use this if data is null, and then you can return an empty.

I moved some duplicate code into a utility class